internal/cmd/io_binary.go

```diff
@@ -105,6 +115,34 @@ func NewBinaryIO(ctx context.Context, id string, uri *url.URL) (_ UpstreamIO, er
 	}, nil
 }
 
+// sanitizePath parses the URL object and returns an absolute path to the logging driver
+func sanitizePath(uri *url.URL) (sanitized string) {
+	path := uri.Path
+	// Convert windows-style slashes to linux-style to clean the path
+	path = strings.Replace(path, "\\", "/", -1)
+	path = filepath.Clean(path)
+	// Path is clean, convert back
+	sanitized = strings.Replace(path, "/", "\\", -1)
+	if sanitized == "." {
+		sanitized = ""
+	}
+
+	// Strip any leading '\\' if the path contains a hard drive in it.
+	if strings.Contains(sanitized, ":\\") {
+		if sanitized[0] == '\\' {
+			// For cleaned paths there should be only one leading '\\'
+			return sanitized[1:]
+		}
+		return sanitized
+	}
+
+	// Absolute path is required, treat "binary://path/to/binary" and "binary:///path/to/binary"
```

Comment:
In my opinion, we should just ignore `url.Host`, and tell users to specify the url as `binary:///path/to/binary` instead. It looks like the containerd runc shim just uses `url.Path`, so I don't think there's any expectation that we need to use `Host` .

Reply:
I think in the initial PR we had some discussion with others about it, but making things work like in upstream sounds reasonable to me